### ISSUE TITLE
Empty nested variables breaking templates when used inside components

### ISSUE THREAD

**Issue by jebax** [maizzle/framework#704]
- Maizzle Version: 4.0.0
- Node.js Version: 16.5.3

Hi,

It'd be really handy if we were able to build templates locally that include nested variables e.g. `page.myVariable.myNestedVariable`. Currently we're having to insert these into our maizzle config when developing locally, e.g:
```node
const config = {
  ...
  myVariable: {
  }
}
```

If `maizzle serve` used `lodash/get` (or similar) for getting template variables that would save us a lot of hassle. Alternatively, if `maizzle serve` supported optional chaining that would also help, but even on new versions of node optional chaining doesn't seem to do anything, the templates still break.

Thanks!

**Comment by cossssmin** [comment]
You can nest variables in YAML Front Matter, is that what you're looking to do?

**Comment by jebax** [comment]
> You can nest variables in YAML Front Matter, is that what you're looking to do?

Not really, although that would fix the problem we'd end up with quite a large amount of front matter including various nested objects and arrays. We don't really want that hanging around in our code since this is only a problem for local development.

**Comment by cossssmin** [comment]
OK, so let me see if I understand, you'd like to write something like this?

```
{{ page.foo?.bar || 'default' }}
```

... so if `foo` is not even defined, it'd output `default`?

**Comment by jebax** [comment]
> OK, so let me see if I understand, you'd like to write something like this?
> 
> ```
> {{ page.foo?.bar || 'default' }}
> ```
> 
> ... so if `foo` is not even defined, it'd output `default`?

Yes exactly, or if you did just `{{ page.foo?.bar }}` you would get the string `undefined`

**Comment by cossssmin** [comment]
That works in `4.0.0` :)

**Comment by jebax** [comment]
> That's exactly how it works in `4.0.0` :)

Ok, weird because I tested yesterday with 4.0.0 and seemed to break, but it could be something else up with a larger template. I'll try with a new template and i'll close this PR if it works. thanks

**Comment by cossssmin** [comment]
I just tested in a fresh new project and simply replaced the contents of `promotional.html` with this:

```html
<extends src="src/layouts/main.html">
  <block name="template">
    {{ page.foo?.bar || 'default' }}
  </block>
</extends>
```

Could be related to outdated dependencies / your yarn problem perhaps?

**Comment by jebax** [comment]
> I just tested in a fresh new project and simply replaced the contents of `promotional.html` with this:
> 
> ```
> <extends src="src/layouts/main.html">
>   <block name="template">
>     {{ page.foo?.bar || 'default' }}
>   </block>
> </extends>
> ```
> 
> Could be related to outdated dependencies / your yarn problem perhaps?

Tried again and it worked, I think it was due to me not deleting `node_modules` before reinstalling like you say! Don't think it's related to that other issue though.

Thanks for your time anyway, I appreciate it

**Comment by jebax** [comment]
Hi, sorry I'm reopening this as I think I've worked out what the actual problem is.

Empty nested variables can't be accessed inside **components**, but inside top-level templates it's fine.

To reproduce:
1. In the starter project, add `{{ page.x.y.z }}` inside `promotional.html` - this will build
2. Add `{{ page.x.y.z }}` inside the `<content>` tag in `v-fill.html` - this will break

Is this intentional? If so, is there a way around it?

**Comment by jebax** [comment]
> Yeah, it's not intentional, just the way posthtml-modules (what we use for components) works.
> 
> A way around it is to use a fallback of some kind, or optional chaining:
> 
> ```
> 
> <content></content>
> 
> {{ page.x?.y.z }}
> ```

Ah ok, thanks for the info!
